idling/v1alpha2/types.go

```diff
@@ -0,0 +1,94 @@
+/*
+Copyright 2018 Red Hat, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+   http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha2
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// NB: slice types *must* be omitempty, otherwise the CRD
+// validator will complain that slices can't be null (since nil slices
+// become null, not empty arrays).
+
+// IdlerSpec defines the desired state of Idler
+type IdlerSpec struct {
+	// WantIdle represents the desired state of idling
+	WantIdle bool `json:"wantIdle"`
+	// TargetScalables contains the collection of scalables that
+	// are idled/unidled together.
+	TargetScalables []CrossGroupObjectReference `json:"targetScalables,omitempty"`
+	// TriggerServiceNames contains the collection of services that shold
+	// trigger unidling.  Their corresponding endpoints objects will be
+	// used to determine whether or not unidling is successful.
+	TriggerServiceNames []string `json:"triggerServiceNames,omitempty"`
+}
+
+// IdlerStatus defines the observed state of Idler
+type IdlerStatus struct {
+	// Idled represents the current state of idling
+	Idled bool `json:"idled"`
```

Comment:
Using a single value when you have multiple potential targets seems questionable.  What happens if you're half-way idled?

Reply:
that's intentional.  I talk about this a bit in the readme (https://github.com/openshift/service-idler#usage-with-network-traffic-based-unidling).  A single idler represents a group of resources that should go up/come down together.  Any given idler is considered idled as soon as one of the resources starts scaling down, and is not considered fully unidled until all of the trigger services have endpoints (this is important for a router-based corner case).